src/inputs/pipe.c

```diff
@@ -302,6 +304,51 @@ parse_progress(struct input_metadata *m, char *progress)
   m->song_length = (end - start) * 10 / 441; // Convert to ms based on 44100
 }
 
+static void
+parse_volume(char *volume)
+{
+  char *value_begin;
+  char *value_next;
+  char *value_end;
+  float airplay_volume;
+
+  if (!(value_begin = strtok_r(volume, ",", &value_next)))
+    return;
+
+  airplay_volume = strtof(value_begin, &value_end);
+
+  if ((errno == ERANGE) || (value_begin == value_end))
+    {
+      char no_number[] = "First token is not a number.";
+      char *failure_detail = errno == ERANGE ? strerror(errno) : no_number;
+      DPRINTF(E_LOG, L_PLAYER, "Invalid airplay volume in string (%s): %s\n", volume, failure_detail);
+      return;
+    }
+
+  if (strcmp(value_next, pipe_metadata_pvol_enabled_suffix) != 0)
+    {
+      DPRINTF(E_DBG, L_PLAYER, "Not applying airplay volume while software volume control is enabled (%s)\n", volume);
+      return;
+    }
+
+  if (airplay_volume > -144.5 && airplay_volume < -143.5)
+    {
+      DPRINTF(E_DBG, L_PLAYER, "Applying airplay volume ('mute', value: %.2f)\n", airplay_volume);
+      player_volume_set(0);
+    }
+  else if (airplay_volume >= -30.0 && airplay_volume <= 0.0)
+    {
+      double local_volume = 100.0 + (airplay_volume / 30.0 * 100.0);
```

Comment:
My code analysis complains that float may not be large enough. Just followed this suggestion to avoid warnings (Neither ram nor performance is an issue here).